.github/workflows/tidy.yml

```diff
@@ -0,0 +1,35 @@
+name: Check with clang-tidy
```

Comment:
This generally should not be separate workflow, but job that runs after one of the builds. Need to discuss the right place :thinking:

Reply:
See https://github.com/eclipse-iceoryx/iceoryx/pull/890#issuecomment-890816831